lib/Makefile

```diff
@@ -112,10 +107,35 @@ clean:
 	@echo Cleaning library completed
 
 
-#------------------------------------------------------------------------
-#make install is validated only for Linux, OSX, kFreeBSD, Hurd and
-#FreeBSD targets
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU FreeBSD))
+#-----------------------------------------------------------------------------
+# make install is validated only for Linux, OSX, BSD, Hurd and Solaris targets
+#-----------------------------------------------------------------------------
+ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS))
+
+ifneq (,$(filter $(shell uname),SunOS))
+INSTALL ?= ginstall
+else
+INSTALL ?= install
+endif
+
+ifneq (,$(filter $(shell uname),OpenBSD FreeBSD NetBSD DragonFly SunOS))
+PREFIX  ?= /usr
```

Comment:
This one is unexpected, though it could be SUN specific :
- `/usr` is for "distribution controlled" software. See it as the target directory for `apt-get install` on Ubuntu for example. The installer is also distribution controlled.
- while `/usr/local` is for "local installation", such as someone downloading lz4 from github, and running locally `make install`.

Therefore, the `Makefile` provided in the project directory should, by default, install to `/usr/local`. Sun (now Oracle) should be in charge of installing lz4 into `/usr` if they see it fit to belong to the distribution.

Now, it could also be that Sun convention is different.
In which case, a link to the place which explains it would be welcomed in the source code.